From 47b926253f00d6599b1f629b2b6463b50ba31c2a Mon Sep 17 00:00:00 2001
From: Richard Zhao <b20223@freescale.com>
Date: Thu, 25 Sep 2008 09:38:53 +0800
Subject: [PATCH] ENGR00092845 wm8350: alloc ASoC machine with struct wm8350

include/linux/regulator/wm8350/wm8350-bus.h:
In struct wm8350, audio is a pointer, but it don't point to a memory allocated.
Change "*audio" to "audio".

Signed-off-by: Richard Zhao <b20223@freescale.com>
---
 include/linux/regulator/wm8350/wm8350-bus.h |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/include/linux/regulator/wm8350/wm8350-bus.h b/include/linux/regulator/wm8350/wm8350-bus.h
index ca22f83..6c638d8 100644
--- a/include/linux/regulator/wm8350/wm8350-bus.h
+++ b/include/linux/regulator/wm8350/wm8350-bus.h
@@ -20,6 +20,7 @@
 #include <linux/regulator/wm8350/wm8350-supply.h>
 #include <linux/regulator/wm8350/wm8350-rtc.h>
 #include <linux/regulator/wm8350/wm8350-wdt.h>
+#include <sound/soc.h>
 
 /* number of WM8350 interrupts */
 #define WM8350_NUM_IRQ	79
@@ -64,7 +65,7 @@ struct wm8350 {
 	struct wm8350_pmic pmic;
 	struct wm8350_wdg wdg;
 	struct wm8350_rtc rtc;
-	struct snd_soc_machine *audio;
+	struct snd_soc_machine audio;
 	struct wm8350_power power;
 	
 	/* irq handlers and workq */
-- 
1.5.4.4

